server/aesdsocket.c

```diff
@@ -228,31 +144,49 @@ void *socket_communication(void *threadp)
     char *heap_buffer_for_write = NULL;
     char *heap_buffer_for_read = NULL;
     long filled_up_buffer_size = 0;
+    long total_buffer_size_for_read_from_file = BUF_SIZE;
     char buf[BUF_SIZE];
     long total_buffer_size_for_write_into_file = BUF_SIZE;
-    long total_buffer_size_for_read_from_file = BUF_SIZE;
-    char *newline_char_ptr;
-    long newline_char_index = 0;
-    long prev_newline_char_index = 0;
-    long end_of_file_index = 0;
+    long total_bytes_read_since_last_newline = 0;
+    long total_bytes_read_from_file = 0;
+    long total_bytes = 0;
+
+    int required_memory = 0;
     //char str[INET_ADDRSTRLEN];
 
     threadParams_t *thread_params = (threadParams_t *)threadp;
+    int new_client_fd = thread_params->sockfd_client;
 
-    
     heap_buffer_for_write = (char *)malloc(sizeof(char) * BUF_SIZE);
     if (heap_buffer_for_write == NULL)
     {
         LOG_ERROR("could not malloc");
         server_socket_state = STATE_EXIT;
     }
-    server_socket_state = STATE_RECEIVE_FROM_CLIENT;
 
-    
+    server_socket_state = STATE_OPENING_FILE;
+
     while (1)
     {
         switch (server_socket_state)
         {
+
+        case STATE_OPENING_FILE:
+
+#ifdef USE_AESD_CHAR_DRIVER
```

Comment:
@rajatchaple not sure why this would need to be different for the `USE_AESD_CHAR_DRIVER` case?